test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_linear.cxx

```diff
@@ -25,88 +25,142 @@
  */
 
 #include <test/t8_gtest_macros.hxx>
-#include <test/t8_geometry/t8_gtest_geometry_macros.hxx>
+#include <test/t8_gtest_custom_assertion.hxx>
 #include <gtest/gtest.h>
 #include <t8_eclass.h>
 #include <t8_cmesh.hxx>
 #include <t8_cmesh/t8_cmesh_examples.h>
 #include <t8_geometry/t8_geometry.h>
 #include <t8_geometry/t8_geometry_implementations/t8_geometry_linear.hxx>
+#include <t8_geometry/t8_geometry_implementations/t8_geometry_linear_axis_aligned.hxx>
 #include <t8_element.h>
 
-t8_cmesh_t
-t8_geometry_testing_tree_from_class (const t8_eclass_t eclass)
-{
-  t8_cmesh_t cmesh;
-  t8_cmesh_init (&cmesh);
-  const int num_vertices = t8_eclass_num_vertices[eclass];
-  double *vertices = T8_ALLOC (double, T8_ECLASS_MAX_DIM *num_vertices);
-  for (int i_vertex = 0; i_vertex < num_vertices; ++i_vertex) {
-    for (int dim = 0; dim < T8_ECLASS_MAX_DIM; ++dim) {
-      vertices[i_vertex * T8_ECLASS_MAX_DIM + dim] = t8_element_corner_ref_coords[eclass][i_vertex][dim];
+class geometry_test: public testing::TestWithParam<std::tuple<int, t8_eclass>> {
+ public:
+  static void
+  SetUpTestSuite ()
+  {
+    seed = time (NULL); /* RNG seed */
+  }
+  static int seed;
+
+ protected:
+  void
+  SetUp () override
+  {
+    const int geom_int = std::get<0> (GetParam ());
+    eclass = std::get<1> (GetParam ());
+    t8_cmesh_init (&cmesh);
+    if (geom_int == T8_GEOMETRY_TYPE_LINEAR_AXIS_ALIGNED
+        && !(eclass == T8_ECLASS_LINE || eclass == T8_ECLASS_QUAD || eclass == T8_ECLASS_HEX)) {
+      GTEST_SKIP ();
+    }
```

Comment:
I would rather just initialize the testsuite only with the line, quad and hex eclass

Reply:
But for the linear geometry we want to test all eclasses.